### PR TITLE
Fix readme, fix publish issue

### LINKED ISSUE #153: empty package is published to maven
when the repo is published, there is nothing in it: https://repo1.maven.org/maven2/xyz/block/tbdex/0.9.0-beta/ that is an empty jar file and pom with no dependencies. It exists in maven central but has no content, so something is broken with build and publish.

### PR DESCRIPTION
**Pull request by jiyoontbd**
closes #153 

also bumping all instances of jackson dependencies to be the same `2.14.2`

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -151,7 +151,7 @@ tasks.dokkaHtmlMultiModule {
 
 publishing {
   publications {
-    create<MavenPublication>("tbdex") {
+    create<MavenPublication>(rootProject.name) {
       groupId = project.group.toString()
       artifactId = name
       description = "Kotlin SDK for tbdex functionality"
```
Comment (jiyoontbd) [inline build.gradle.kts:157]: thinking i should somehow enable us to package all 3 projects exported under 1 name - `tbdex`. looking into shadow jars

File: build.gradle.kts [hunk 2]
```diff
@@ -20,6 +21,9 @@ repositories {
 }
 
 dependencies {
+  api(project(":protocol"))
```
Comment (jiyoontbd) [inline build.gradle.kts:24]: we needed these to actually get all 3 subprojects when we pull in `tbdex` as a dependency

File: build.gradle.kts [hunk 3]
```diff
@@ -101,11 +112,13 @@ subprojects {
       }
     }
 
-    signing {
-      val signingKey: String? by project
-      val signingPassword: String? by project
-      useInMemoryPgpKeys(signingKey, signingPassword)
-      sign(publishing.publications[publicationName])
+    if (!project.hasProperty("skipSigning") || project.property("skipSigning") != "true") {
```
Comment (jiyoontbd) [inline build.gradle.kts:115]: adding this allows you to build artifacts locally even without a signing key and password like this

`./gradlew publishToMavenLocal -PskipSigning=true -Pversion={your-local-version-name}`

File: build.gradle.kts (hunk 1 above)
Comment (jiyoontbd) [reply build.gradle.kts:170]: nvm found how to do it [here](https://github.com/TBD54566975/tbdex-kt/pull/152/files#r1483160314)

File: README.md [hunk 4]
```diff
@@ -10,17 +10,19 @@ This repo contains 2 jvm packages:
...
-
 ```kotlin
 repositories {
+  mavenCentral()
   maven("https://jitpack.io")
   maven("https://repo.danubetech.com/repository/maven-public/")
 }
 
 dependencies {
+  implementation("com.github.TBD54566975:tbdex:0.4.0-beta")
```
Comment (mistermoe) [inline README.md:24]: Do we need to update this to reflect the new group id and version?

**Review by mistermoe — APPROVED** [review]
Awesome work @jiyoontbd !

File: README.md (hunk 4 above)
Comment (jiyoontbd) [reply README.md:24]: oh yea. i missed this in merge conflict fix. ty!

**Pull request merged**